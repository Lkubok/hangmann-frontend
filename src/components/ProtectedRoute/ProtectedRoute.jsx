import React, { Component } from "react";
import { Route, Redirect } from "react-router-dom";

export default class ProtectedRoute extends Component {
  render() {
    const { isAuth } = this.props;
    return isAuth ? (
      <Route path={this.props.path} component={this.props.component} />
    ) : (
      <Redirect to="/user" />
    );
  }
}
