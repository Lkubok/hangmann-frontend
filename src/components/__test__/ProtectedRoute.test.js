import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { ProtectedRoute } from "../ProtectedRoute/ProtectedRoute";
Enzyme.configure({ adapter: new Adapter() });

describe("ProtectedRoute component", () => {
  beforeEach(() => {});
  it("Should match snapshot", () => {
    const wrapper = shallow(<ProtectedRoute />);
    expect(wrapper).toMatchSnapshot();
  });
});
