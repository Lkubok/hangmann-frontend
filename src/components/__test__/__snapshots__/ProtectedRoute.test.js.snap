// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute component Should match snapshot 1`] = `
<Redirect
  to="/user"
/>
`;
